src/matchers/excel.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+const { OneDrive } = require('@adobe/helix-onedrive-support');
+const XLSX = require('xlsx');
+
+async function extract(url, params, log = console) {
+  const {
+    AZURE_WORD2MD_CLIENT_ID: clientId,
+    AZURE_HELIX_USER: username,
+    AZURE_HELIX_PASSWORD: password,
+  } = params;
+
+  try {
+    const drive = new OneDrive({
+      clientId,
+      username,
+      password,
+      log,
+    });
+
+    const driveItem = await drive.getDriveItemFromShareLink(url);
+    const xlsxbuffer = await drive.downloadDriveItem(driveItem);
+
+    const workbook = XLSX.read(xlsxbuffer, { type: 'buffer' });
```

Comment:
probably better to use the excel api:
https://docs.microsoft.com/en-us/graph/api/resources/excel?view=graph-rest-1.0

eg: https://docs.microsoft.com/en-us/graph/api/worksheet-get?view=graph-rest-1.0&tabs=http#example

Reply:
I tried, but ran into this error: https://stackoverflow.com/questions/50216300/could-not-obtain-wac-access-token

Reply:
but here it works:
https://github.com/adobe/helix-theblog-scanner/blob/master/src/handlers/ExcelHandler.js